### ISSUE TITLE
reserve -> AMM integration test / story

### ISSUE THREAD

**Issue by dckc** [Agoric/Agoric#5298]
## What is the Problem Being Solved?

From the draft whitepaper:

> 2.4 Reserve Pool
> The Reserve pool ... is used to provide liquidity to Kinetic for liquidation.

_opened on behalf of @rowgraus_

## Description of the Design

When the BLDer DAO elects the governance committee, it sets up an [econCommitteeCharter.js](https://github.com/Agoric/agoric-sdk/blob/3924-gov-vaults/packages/run-protocol/src/econCommitteeCharter.js) contract that allows the committee to vote on all and only the powers they should have. In particular, it allows them to vote to invoke `addLiquidityToAmmPool` on the reserve contract to move assets to the AMM, as in the unit test below:

https://github.com/Agoric/agoric-sdk/blob/9b5565d197187223fe22786aeddfd800263278aa/packages/run-protocol/test/reserve/test-reserve.js#L147-L155

## Security Considerations

Granting the committee all and only the powers they should have involves careful attenuation of some very powerful objects, such as the creator facets of the reserve, vault factory, etc.

## Test Plan

Use devnet BLDer DAO governance to test this flow, as in, for example,
https://testnet.ping.pub/agoric-devnet/gov/9
Voting Time | 2022-05-02 20:12 - 2022-05-02 20:22

**Comment by dckc** [comment]
Ugh! The IbcAMM pool is stuck due to #5293 , and that's the only issuer we introduced the reserve to in the recent BLDer DAO vote.

I just about have the details of an econ committee worked out, but it's not going to be able to actually put liquidity in.

**Comment by dckc** [comment]
I can't even get as far as calling for votes.

I get `undefined is a undefined but must be a bigint or a number` when I try to `voteOnReserveApiInvocation`.
details: https://gist.github.com/dckc/2b0d26f81c8c838d4641ad8edbaff650

and there are no stack-traces available.

**Comment by Chris-Hibbert** [comment]
I've often seen `undefined is a undefined but must be a bigint or a number` in reference to the deadline when starting a vote, but I haven't been able to narrow it down in this case.  Any chance your deadline wasn't a bigint?

**Comment by dckc** [comment]
> Any chance your deadline wasn't a bigint?

I have ruled out that problem to the best of my ability, but I can't say there's no chance.

I'm confident that the deploy script passes a bigint `await E(charterAPI).voteOnReserveApiInvocation(amounts,deadline)` because I logged it on the line before: `new Date(Number(deadline) * 1000).toISOString()` -> `2022-05-05T06:11:27.000Z`.

I stared at [econCommitteeCharter.js](https://github.com/Agoric/agoric-sdk/blob/3924-gov-vaults/packages/run-protocol/src/econCommitteeCharter.js) a _lot_ and couldn't find any mismatches between that and the unit test cited above.

**Comment by Chris-Hibbert** [comment]
Yeah, I saw `new Date(Number(deadline) * 1000).toISOString()`, and that didn't convince me that `deadline` was a bigint.

**Comment by dckc** [comment]
oh... you mean not only could `undefined` prompt those symptoms, but also a `number`? hm...

**Comment by dckc** [comment]
it's defined as `const deadline = currentTimestamp + 3n;`. I'm pretty sure that makes a bigint.

If `currentTimestamp` were a number, I'm pretty sure that would throw:

```
> const deadline = 3 + 3n
Uncaught TypeError: Cannot mix BigInt and other types, use explicit conversions
```

JavaScript is full of surprises, so we could get `undefined` if `currentTimestamp` were a string...

```
> const deadline = 'x' + 3n
undefined
```

but that wouldn't log as `2022-05-05T06:11:27.000Z`

**Comment by Tartuffo** [comment]
We can test when RC0 is on devnet, so not a blocker for RC0 but is a blocker for Launch.
